arch/arm/core/cortex_m/CMakeLists.txt

```diff
@@ -9,3 +9,12 @@ zephyr_library_sources(
   nmi.c
   exc_manage.c
   )
+
+add_linker_snippet_ifdef(CONFIG_SW_VECTOR_RELAY RAM_SECTIONS
+  "/* Reserved 4 bytes to save vector table base address */
+  SECTION_PROLOGUE(.vt_pointer,(NOLOAD),)
+  {
+          *(.vt_pointer_section)
+          *(\".vt_pointer_section.*\")
+  }"
```

Comment:
Thoughts on having the LD script code in .ld files instead?

Benefits as I see it;

- Easy to find LD script code, (look for .ld files).
- Easier for an editor to help with formatting and syntax highlighting.
- User does not need to know which symbols need to be escaped.
- Consistent with how we add other files, e.g. c files.

Reply:
Sure, I see no big reasons not to. So the Cmake would look something like this maybe?

    zephyr_linker_sources(<location> <files>)
e.g:

    zephyr_linker_sources(RODATA
      ${CMAKE_CURRENT_LIST_DIR}/mylinkerscript.ld
    )

Reply:
yes

Reply:
check out how zephyr_sources is implemented to see how to get rid of the CMAKE_CURRENT_LIST_DIR.

Reply:
Fixed in upstream PR